File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        invert(root);
        return root;
    }

    private void invert(TreeNode node){
        if(node == null)
            return;
        TreeNode leftNode = node.left;
        TreeNode rightNode = node.right;

        node.left = rightNode;
        node.right = leftNode;

        invert(node.left);
        invert(node.right);

    }
}

File: Data Structures & Algorithms/sort-array-by-parity/submission-0.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        left = 0
        right = len(nums)-1

        while left < right:
            while left < right and nums[left] % 2 == 0:
                left += 1
            while left < right and nums[right] % 2 != 0:
                right -= 1
            
            if left < right:
                t = nums[right]
                nums[right] = nums[left]
                nums[left] = t
                left += 1
                right -= 1
        
        return nums